PeakNetwork/Core/NetworkOperation.swift

```diff
@@ -66,34 +80,139 @@ open class NetworkOperation: RetryingOperation<NetworkResponse>, ConsumesResult
     /// - Returns: A URLSessionTask, or nil.
-    open func createTask(with request: URLRequest, using session: Session) -> URLSessionTask? {
-        let task = session.dataTask(with: request) { [weak self] data, response, error in
+    open func createTask(with resource: Resource<O>, using session: Session) -> URLSessionTask? {
+        let task = session.dataTask(with: resource.request) { [weak self] data, response, error in
             guard let strongSelf = self else { return }
             guard !strongSelf.isCancelled else { return strongSelf.finish() }
 
             if let error = error {
                 strongSelf.output = Result { throw error }
             } else if let httpResponse = response as? HTTPURLResponse {
                 if httpResponse.statusCodeValue.isSuccess {
-                    strongSelf.output = .success((data, httpResponse))
-                } else {
                     strongSelf.output = Result {
-                        throw ServerError.error(code: httpResponse.statusCodeValue, data: data, response: httpResponse)
+                        Response(data: data, urlResponse: httpResponse, parsed: try resource.parse(data))
                     }
+                } else {
+                    strongSelf.output = .failure(ServerError.error(code: httpResponse.statusCodeValue, data: data, response: httpResponse))
                 }
             } else {
-                strongSelf.output = Result {
-                    throw ServerError.unknownResponse
-                }
+                strongSelf.output = .failure(ServerError.unknownResponse)
             }
             strongSelf.finish()
         }
         
-        if #available(iOS 11.0, *), #available(tvOS 11.0, *) {
+        if #available(iOS 11.0, *) {
             progress.addChild(task.progress, withPendingUnitCount: progress.totalUnitCount)
         }
         
         return task
     }
 }
 
+extension NetworkOperation {
+    /// Unwrap the Response and return a result containing only the
+    /// parsed data, discarding the network response information.
+    public var unwrapped: MapOperation<Response<O>, O> {
+        return passesResult(to: BlockMapOperation<Response<O>, O> { input in
+            switch (input) {
+            case .success(let response):
+                return .success(response.parsed)
+            case .failure(let error):
+                return .failure(error)
+            }
+        })
+    }
+}
+
+/// The outcome of the requests. The object provided with the `Resource`
+/// is associated with the the `Response` or `Error`.
+public struct MultipleResourceOutcome<E, O> {
+    public let successes: [(object: E, response: Response<O>)]
+    public let failures: [(object: E, error: Error)]
+}
+
+/// Perform a series of network requests on an internal queue and aggregate the results.
+open class MultipleResourceNetworkOperation<E, O>: ConcurrentOperation, ConsumesResult, ProducesResult {
```

Comment:
I understand the utility of this operation, but feel it may be too specific or niche to include in the framework. Is the Peak Framework not meant to provide a set of building blocks?

Reply:
This is true. Where do we keep it, then? Just as a snippet somewhere?

Reply:
Since it was originally written for a specific internal problem, maybe we just stick it in our private 'catch-all' repo.